pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -401,18 +400,21 @@ private void handleBrokerConnected(DirectProxyHandler directProxyHandler, Comman
     }
 
     private void connectToBroker(InetSocketAddress brokerAddress) {
-        checkState(ctx.executor().inEventLoop(), "This method should be called in the event loop");
+        assert ctx.executor().inEventLoop();
```

Comment:
"assert" is not enabled in production
I am not sure that we enable Java assertion in tests

Reply:
Correct. I had opened https://github.com/apache/pulsar/pull/19653, but @lhotari pointed out that it is already enabled by surefire.